### ISSUE TITLE
Documentation bug on PCAMatrix

### ISSUE THREAD

**Issue by dbickson** [facebookresearch/faiss#2506]
# Summary

# Platform

OS: 

Faiss version: 

Installed from:  

Faiss compilation options: 

Running on:
- [x] CPU
- [ ] GPU

Interface: 
- [x] C++
- [ ] Python

# Reproduction instructions
C++ code documentation under https://faiss.ai/cpp_api/struct/structfaiss_1_1PCAMatrix.html
says for apply(): `apply the random rotation, return new allocated matrix`. 
Should be apply the PCA dimensionality reduction, since bool random_rotation = false in the default PCAMatrix constructor and it still applies the PCA fit_transform (as it is called in scikit-learn where it is explained better: "Fit the model with X and apply the dimensionality reduction on X.")

**Comment by mdouze** [comment]
right, this is an inherited comment from the parent class, that is also inaccurate

https://github.com/facebookresearch/faiss/blob/main/faiss/VectorTransform.h#L46
